Code/RDGeneral/ConcurrentQueue.h

```diff
@@ -0,0 +1,102 @@
+#ifdef RDK_THREADSAFE_SSS
+#ifndef CONCURRENT_QUEUE
+#define CONCURRENT_QUEUE
+#include <condition_variable>
+#include <queue>
+#include <thread>
+
+namespace RDKit {
+template <class E>
+class ConcurrentQueue {
+ private:
+  size_t capacity;
```

Comment:
to make the rest of the code easier to read, please prefix class data members with `d_`. i.e. `d_capacity`, `d_done`, etc.